src/coreclr/inc/corinfo.h

```diff
@@ -2082,9 +2081,16 @@ class ICorStaticInfo
             CORINFO_CLASS_HANDLE        classHnd
             ) { return false; }
 
-    // return the unmanaged calling convention for a PInvoke
-    virtual CorInfoUnmanagedCallConv getUnmanagedCallConv(
-            CORINFO_METHOD_HANDLE       method
+    // return the entry point calling convention for any of the following
+    // - a P/Invoke
+    // - a method marked with UnmanagedCallersOnly
+    // - a function pointer with the CORINFO_CALLCONV_UNMANAGED calling convention.
+    // May return CorInfoCallConvExtension::Managed when the method or call site doesn't
+    // use an unmanaged calling convention as an entry point.
+    virtual CorInfoCallConvExtension getUnmanagedCallConv(
+            CORINFO_METHOD_HANDLE       method,
+            CORINFO_SIG_INFO*           callSiteSig,
+            bool*                       pSuppressGCTransition = NULL /* OUT */
```

Comment:
Does this really need to be a optional? Is there a case where not knowing this is okay - seems scary to ask for a convention but not need to know to suppress the transition.

Reply:
Yes. When we ask for the unmanaged calling convention of a Reverse P/Invoke stub or an UnmanagedCallersOnly function, we don't need to know to suppress the GC transition. I can make this non-optional if we prefer.